src/Calculator/Utils/JsonUtils.cs

```diff
@@ -0,0 +1,330 @@
+        public IEnumerable<CalcManagerTokenAlias> Tokens
+        {
+            get => Value.Tokens.Select(x => new CalcManagerTokenAlias(x));
+            set => Value.Tokens = value.Select(Helpers.MapHistoryToken).ToList();
+        }
+        [JsonPropertyName("c")]
+        public IEnumerable<ICalcManagerIExprCommandAlias> Commands
+        {
+            get => Value.Commands.Select(Helpers.MapCommandAlias);
+            set => Value.Commands = value.Select(Helpers.MapCommandAlias).ToList();
+        }
+        [JsonPropertyName("e")]
+        public string Expression
+        {
+            get => Value.Expression;
+            set => Value.Expression = value;
+        }
+        [JsonPropertyName("r")]
+        public string Result
+        {
+            get => Value.Result;
+            set => Value.Result = value;
+        }
+
+        public CalcManagerHistoryItemAlias() => Value = new CalcManagerHistoryItem();
+        public CalcManagerHistoryItemAlias(CalcManagerHistoryItem value) => Value = value;
+    }
+
+    internal class CalcManagerSnapshotAlias
+    {
+        [JsonIgnore]
+        public CalcManagerSnapshot Value;
+
+        [JsonPropertyName("h")]
+        public IEnumerable<CalcManagerHistoryItemAlias> HistoryItems
+        {
+            get => Value.HistoryItems.Select(x => new CalcManagerHistoryItemAlias { Value = x });
+            set => Value.HistoryItems = value.Select(x => new CalcManagerHistoryItem
+            {
+                Tokens = x.Tokens.Select(Helpers.MapHistoryToken).ToList(),
+                Commands = x.Commands.Select(Helpers.MapCommandAlias).ToList(),
+                Expression = x.Expression,
+                Result = x.Result
+            }).ToList();
+        }
+
+        public CalcManagerSnapshotAlias() => Value = new CalcManagerSnapshot();
+        public CalcManagerSnapshotAlias(CalcManagerSnapshot value) => Value = value;
+    }
+
+    internal class PrimaryDisplaySnapshotAlias
+    {
+        [JsonIgnore]
+        public PrimaryDisplaySnapshot Value;
+
+        [JsonPropertyName("d")]
+        public string DisplayValue
+        {
+            get => Value.DisplayValue;
+            set => Value.DisplayValue = value;
+        }
+        [JsonPropertyName("e")]
+        public bool IsError
+        {
+            get => Value.IsError;
+            set => Value.IsError = value;
+        }
+
+        public PrimaryDisplaySnapshotAlias() => Value = new PrimaryDisplaySnapshot();
+        public PrimaryDisplaySnapshotAlias(PrimaryDisplaySnapshot value) => Value = value;
+    }
+
+    internal class ExpressionDisplaySnapshotAlias
+    {
+        [JsonIgnore]
+        public ExpressionDisplaySnapshot Value;
+
+        [JsonPropertyName("t")]
+        public IEnumerable<CalcManagerTokenAlias> Tokens
+        {
+            get => Value.Tokens.Select(x => new CalcManagerTokenAlias(x));
+            set => Value.Tokens = value.Select(Helpers.MapHistoryToken).ToList();
```

Comment:
Remove "History" around "Token" in other places as well?

Reply:
updated